ResearchKit/Common/ORKTintedImageView.m

```diff
@@ -30,56 +30,67 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-    CGContextRef context     = UIGraphicsGetCurrentContext();
-    CGContextSetBlendMode(context, kCGBlendModeNormal);
-    CGContextSetAlpha(context, 1);
-    
-    CGRect r = (CGRect){{0,0},image.size};
-    CGContextBeginTransparencyLayerWithRect(context, r, NULL);
-    [tintColor setFill];
-    [image drawInRect:r];
-    UIRectFillUsingBlendMode(r, kCGBlendModeSourceIn);
-    CGContextEndTransparencyLayer(context);
-    
-    UIImage *outputImage = UIGraphicsGetImageFromCurrentImageContext();
-    UIGraphicsEndImageContext();
+    UIImage *outputImage = nil;
+    if (image && tintColor && scale > 0) {
+        UIGraphicsBeginImageContextWithOptions(image.size, NO, scale);
+        CGContextRef context     = UIGraphicsGetCurrentContext();
+        CGContextSetBlendMode(context, kCGBlendModeNormal);
+        CGContextSetAlpha(context, 1);
+        
+        CGRect r = (CGRect){{0,0},image.size};
+        CGContextBeginTransparencyLayerWithRect(context, r, NULL);
+        [tintColor setFill];
+        [image drawInRect:r];
+        UIRectFillUsingBlendMode(r, kCGBlendModeSourceIn);
+        CGContextEndTransparencyLayer(context);
+        
+        outputImage = UIGraphicsGetImageFromCurrentImageContext();
+        UIGraphicsEndImageContext();
+    }
     return outputImage;
 }
 
 @implementation ORKTintedImageView {
     UIImage *_originalImage;
+    UIImage *_tintedImage;
+    UIColor *_appliedTintColor;
+    CGFloat _appliedScaleFactor;
 }
 
 - (void)setShouldApplyTint:(BOOL)shouldApplyTint {
     _shouldApplyTint = shouldApplyTint;
-    
     self.image = _originalImage;
 }
 
 - (UIImage *)imageByTintingImage:(UIImage *)image {
-    UIImageRenderingMode renderMode = [image renderingMode];
-    BOOL isAnimatedImage = [[image images] count] > 1;
-    
-    // If this is a single image, and it has an automatic rendering mode, and we are supposed to be tinting,
-    // then get a version with tint.
-    if (!isAnimatedImage && renderMode == UIImageRenderingModeAutomatic && _shouldApplyTint) {
-        image = [image imageWithRenderingMode:(UIImageRenderingModeAlwaysTemplate)];
-    // If this is an animated image, and it is a fixed template rendering mode, or, it is an automatic
-    // rendering mode and we are supposed to be tinting, then get a version with tint.  We have to
-    // manually apply the tint to animated images, even if they are in a fixed template rendering mode,
-    // due to:
-    // <rdar://problem/19792197>
-    } else if (isAnimatedImage && (renderMode == UIImageRenderingModeAlwaysTemplate || (renderMode == UIImageRenderingModeAutomatic && _shouldApplyTint))) {
-            NSMutableArray *images = [NSMutableArray array];
-            UIColor *tintColor = [self tintColor];
-            for (UIImage *im in image.images) {
-                [images addObject:ORKImageByTintingImage(im, tintColor, self.contentScaleFactor)];
+    if (image && (image.renderingMode == UIImageRenderingModeAlwaysTemplate
```

Comment:
Don't you think that this one reads better with the indent? I feel that you can more naturally read which conditions make `image` to be changed.